app.js

```diff
@@ -0,0 +1,330 @@
+  if (mainScreen.innerHTML != "") {
+    let memoryClear = document.getElementById("memory-clear");
+    let memoryRecall = document.getElementById("memory-recall");
+    if (storedMemoryData != null) {
+      storedMemoryData.push(mainScreen.innerHTML);
+      localStorage.setItem("calcmemory", JSON.stringify(storedMemoryData));
+      memoryClear.setAttribute("class", "btn");
+      memoryRecall.setAttribute("class", "btn");
+    } else {
+      memoryItems.push(mainScreen.innerHTML);
+      localStorage.setItem("calcmemory", JSON.stringify(memoryItems));
+      memoryClear.setAttribute("class", "btn");
+      memoryRecall.setAttribute("class", "btn");
+    }
+  }
+}
+
+function memoryPlusSubtract(clicked_id) {
+  const storedMemoryData = JSON.parse(localStorage.getItem("calcmemory"));
+  if (mainScreen.innerHTML != "") {
+    if (storedMemoryData != null) {
+      let lastItems = storedMemoryData.length - 1;
+      switch (clicked_id) {
+        case "memory-plus":
+          let replacePlus = eval(storedMemoryData[lastItems] + "+" + mainScreen.innerHTML);
+          storedMemoryData[lastItems] = replacePlus;
+          localStorage.setItem("calcmemory", JSON.stringify(storedMemoryData));
+          break;
+        case "memory-subtract":
+          let replaceSubtract = eval(storedMemoryData[lastItems] + "-" + mainScreen.innerHTML);
+          storedMemoryData[lastItems] = replaceSubtract;
+          localStorage.setItem("calcmemory", JSON.stringify(storedMemoryData));
+          break;
+        default:
+          localStorage.removeItem("calcmemory");
+          checkMemory();
+          break;
+      }
+    } else {
+      return false;
+    }
+  }
+}
+
+function memoryRecall() {
+  let retrievedMemoryData = localStorage.getItem("calcmemory");
+  let memoryData = JSON.parse(retrievedMemoryData);
+  mainScreen.innerHTML = memoryData[memoryData.length - 1];
+}
+
+function checkMemory() {
+  const storedMemoryData = JSON.parse(localStorage.getItem("calcmemory"));
+  let memoryClear = document.getElementById("memory-clear");
+  let memoryRecall = document.getElementById("memory-recall");
+  if (storedMemoryData == null) {
+    memoryClear.className += " disabled";
+    memoryRecall.className += " disabled";
+  } else {
+    memoryClear.setAttribute("class", "btn");
+    memoryRecall.setAttribute("class", "btn");
+  }
+}
+//Memory Operations End
+
+function getMathFunctionValue(clicked_id) {
+  switch (clicked_id) {
+    case "rand":
+      displayOutput(Math.random());
+      break;
+    case "floor":
+      childScreen.innerHTML = `floor(${mainScreen.innerHTML})`;
+      displayOutput(Math.floor(mainScreen.innerHTML));
+      break;
+    case "ceil":
+      childScreen.innerHTML = `ceil(${mainScreen.innerHTML})`;
+      displayOutput(Math.ceil(mainScreen.innerHTML));
+      break;
+  }
+}
+
+function getTrigonometryValue(clicked_id) {
+  switch (clicked_id) {
+    case "sin":
+      childScreen.innerHTML = `sin(${mainScreen.innerHTML})`;
```

Comment:
I want reason why are you using tiled sign(`)?

Reply:
I used back-tics because I did string interpolation "${mainScreen.innerHTML}", it will display the variable value in a string. Here string interpolation is a feature of template literals, so template literals are enclosed using back-tics.